### PR TITLE
8306652: Open source AWT MenuItem related tests 

### PR DESCRIPTION
**Pull request by kumarabhi006**
This PR is to open source MenuItem AWT tests.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306652](https://bugs.openjdk.org/browse/JDK-8306652): Open source AWT MenuItem related tests

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [3d07d952](https://git.openjdk.org/jdk/pull/13589/files/3d07d952fafcc8f4af505c7a7b9d3041ea6f053d)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [3d07d952](https://git.openjdk.org/jdk/pull/13589/files/3d07d952fafcc8f4af505c7a7b9d3041ea6f053d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13589/head:pull/13589` \
`$ git checkout pull/13589`

Update a local copy of the PR: \
`$ git checkout pull/13589` \
`$ git pull https://git.openjdk.org/jdk.git pull/13589/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13589`

View PR using the GUI difftool: \
`$ git pr show -t 13589`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13589.diff">https://git.openjdk.org/jdk/pull/13589.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13589#issuecomment-1518252710)

### CODE REVIEW

File: test/jdk/java/awt/MenuItem/MenuSetLabelTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4261935
+  @summary Menu display problem when changing the text of the menu(window 98)
+  @key headful
+  @run main/timeout=600 MenuSetLabelTest
```
Comment (mrserb) [inline test/jdk/java/awt/MenuItem/MenuSetLabelTest.java:29]: Why the timeout is so big?

File: test/jdk/java/awt/MenuItem/MenuSetLabelTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+class Menu1 extends Frame {
+
+    String s1 = new String("short");
+    String s2 = new String("This is a long string");
+    String s3 = new String("Menu Item string");
+
+    MenuBar mb1 = new MenuBar();
+    Menu f = new Menu(s1);
+    Menu m = new Menu(s1);
+    boolean  flag = true;
```
Comment (turbanoff) [inline test/jdk/java/awt/MenuItem/MenuSetLabelTest.java:82]:
```suggestion
    boolean flag = true;
```

File: test/jdk/java/awt/MenuItem/MenuSetLabelTest.java (hunk 1 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/MenuItem/MenuSetLabelTest.java:29]: I have kept it as it was in the test before conversion.

Comment (kumarabhi006) [reply test/jdk/java/awt/MenuItem/MenuSetLabelTest.java]: Ok. Updated.

File: test/jdk/java/awt/MenuItem/EnableTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    public static void main(String[] args) throws Exception {
+        EnableTest test = new EnableTest();
+        test.start();
+    }
+
+    public void start() throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                frame = new Frame();
```
Comment (prrace) [inline test/jdk/java/awt/MenuItem/EnableTest.java:51]: Can we please have a meaningful Frame title here ? "EnableTest" perhaps ?

File: test/jdk/java/awt/MenuItem/MenuSetLabelTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    public static void main(String[] args) throws Exception {
+        MenuSetLabelTest test = new MenuSetLabelTest();
+        test.start();
+    }
+
+    public void start() throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                f = new Menu1();
```
Comment (prrace) [inline test/jdk/java/awt/MenuItem/MenuSetLabelTest.java:51]: Menu1("MenuSetLabelTest")

File: test/jdk/java/awt/MenuItem/SetStateTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    Frame frame;
+    public static void main(String[] args) throws Exception {
+        SetStateTest test = new SetStateTest();
+        test.start();
+    }
+
+    public void start () throws Exception {
+        try {
+            EventQueue.invokeAndWait(() -> {
+                frame = new Frame();
```
Comment (prrace) [inline test/jdk/java/awt/MenuItem/SetStateTest.java:49]: Frame("SetStateTest")

**Review by prrace — CHANGES_REQUESTED** [review]

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/MenuItem/MenuSetLabelTest.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    Menu f = new Menu(s1);
+    Menu m = new Menu(s1);
+    boolean flag = true;
+
+    public Menu1()
+    {
+        for (int i=0;i<5;i++)
+            m.add(new MenuItem(s3));
+        for (int i=0;i<10;i++)
+            f.add(new MenuItem(s3));
```
Comment (prsadhuk) [inline test/jdk/java/awt/MenuItem/MenuSetLabelTest.java:90]: please add a space before and after operator as in i = 0 for all these

**Review by prsadhuk — APPROVED** [review]

**Pull request closed without merging**